src/prefect/cli/deploy.py

```diff
@@ -1090,28 +1094,66 @@ def _pick_deploy_configs(deploy_configs, names, deploy_all, ci=False):
         # and we are not in interactive mode
         return deploy_configs
     elif len(names) >= 1:
-        # Return all deployment configurations with the given names
-        matched_deploy_configs = [
-            deploy_config
-            for deploy_config in deploy_configs
-            if deploy_config.get("name") in names
-        ]
-        unfound_names = set(names) - {
+        matched_deploy_configs = []
+        deployment_names = []
+        for name in names:
+            # if flow-name/deployment-name format
+            if "/" in name:
+                flow_name, deployment_name = name.split("/")
+                flow_name = flow_name.replace("-", "_")
+                matched_deploy_configs += [
+                    deploy_config
+                    for deploy_config in deploy_configs
+                    if deploy_config.get("name") == deployment_name
+                    and (deploy_config.get("entrypoint") or "").split(":")[1]
+                    == flow_name
```

Comment:
I think you'd want to use `-1` here as the index, otherwise the fallback of `""` will cause an `IndexError`.

```suggestion
                    and deploy_config.get("entrypoint", "").split(":")[-1] == flow_name
```

Reply:
oh yes, great point!